### DISCUSSION TITLE
How to set width = max-content?

### DISCUSSION THREAD

**Discussion by WesleyYue** [csandman/chakra-react-select#246] [Q&A]
[In the vanilla react-select, you could get a automatically sized component with](https://github.com/JedWatson/react-select/issues/4201#issuecomment-874098561):

```
              menu: (base) => ({
                ...base,
                width: "max-content",
                minWidth: "100%",
              }),
```

but this doesn't work in chakra-react-select. What do I need to do differently?

**Comment by csandman** [comment]
This appears to work perfectly to me!

Here's a functional example: https://codesandbox.io/s/jjimj0?file=/app.tsx

The one thing to make sure of is that you're using the `chakraStyles` prop instead of the original `styles` prop.

```tsx
import React, { useState } from "react";
import { Container, FormControl, FormLabel } from "@chakra-ui/react";
import { Select, OptionBase } from "chakra-react-select";
import countries from "./data/countries";

interface CountryOption extends OptionBase {
  label: string;
  value: string;
}

const App: React.FC = () => {
  const [selectedColors, setSelectedColors] = useState<
    readonly CountryOption[]
  >([]);

  return (
    <Container mb={16} maxW="sm">
      <FormControl p={4}>
        <FormLabel>chakra-react-select automatically sized menu</FormLabel>
        <Select
          isMulti
          name="countries"
          options={countries}
          placeholder="Select some countries..."
          closeMenuOnSelect={false}
          value={selectedColors}
          onChange={setSelectedColors}
          chakraStyles={{
            menu: (base) => ({
              ...base,
              width: "max-content",
              minWidth: "100%",
            }),
          }}
        />
      </FormControl>
    </Container>
  );
};

export default App;
```

<img width="438" alt="image" src="https://user-images.githubusercontent.com/9214195/221652406-5a7a4328-6197-4f76-bf58-7a468f68c669.png">

**Reply by WesleyYue** [reply]
Sorry, when I say auto size, I mean the entire component, not just the menu. Please see this [codesandbox](https://codesandbox.io/s/chakra-react-select-auto-sized-menu-forked-ew7khq?file=/app.tsx). In this, the chakra-react-select completely collapses when wrapped with a max-content parent, whereas the react-select sizes the entire component automatically.

```
      <Box width="max-content">
        <Select
          name="countries"
          options={countries}
          placeholder="Select some countries..."
          closeMenuOnSelect={false}
          value={selectedColors}
          onChange={setSelectedColors}
          variant="unstyled"
          chakraStyles={{
            menu: (base) => ({
              ...base,
              width: "max-content",
              minWidth: "100%"
            })
          }}
        />
      </Box>

      <Box width="max-content">
        <ReactSelect
          name="countries"
          options={countries}
          placeholder="Select some countries..."
          closeMenuOnSelect={false}
          value={selectedColors}
          onChange={setSelectedColors}
          styles={{
            menu: (base) => ({
              ...base,
              width: "max-content",
              minWidth: "100%"
            })
          }}
        />
      </Box>
```
![image](https://user-images.githubusercontent.com/5280818/221719010-63153428-2e28-4b27-9b99-ea452758a56b.png)

**Reply by csandman** [reply]
Ah, I see what you're saying. I believe I found a style mismatch between some of the subcomponents of CRS and the original RS. Give me a bit to check out the styles, and I'll most likely push a new version to realign them. If not, I'll find a way to do it with custom styles.

**Reply by csandman** [reply]
Alright, I figured out where the discrepancies were and just pushed up a new version, [`v4.6.0`](https://github.com/csandman/chakra-react-select/releases/tag/v4.6.0)!

Here's a complete functional example on the new version demonstrating that both packages behave in the same way: https://codesandbox.io/s/luczw3?file=/app.tsx

<img width="389" alt="image" src="https://user-images.githubusercontent.com/9214195/222028908-457a8286-fd73-4969-b683-6671b728b5f4.png">

**Reply by WesleyYue** [reply]
awesome, thank you so much!

**Comment by Dregorio1** [comment]
@csandman Hello, and how in new version I can have old style? I want to have fixed width
